plutus-ledger/src/Ledger/Tx/Internal.hs

```diff
@@ -61,9 +61,10 @@ import Prettyprinter (Pretty (..), hang, viaShow, vsep, (<+>))
 
 -- | The type of a transaction input.
 data TxInType =
-      ScriptAddress !(Either (Versioned Validator) (Versioned TxOutRef)) !Redeemer !Datum
+      ScriptAddress !(Either (Versioned Validator) (Versioned TxOutRef)) !Redeemer !(Maybe Datum)
       -- ^ A transaction input that consumes (with a validator) or references (with a txOutRef)
       -- a script address with the given the redeemer and datum.
+      -- Datum is optional in case the input references an inline datum.
```

Comment:
Mmm... What do you mean here _exactly_?

Do you actually mean "Datum is optional if the input refers to a script output which contains an inline datum"?